### PR TITLE
User can specify key manticore options which are loaded with sensible defaults

### PR DESCRIPTION
**Pull request by kokrui**
As per https://github.com/trailofbits/MUI/issues/29

Adds a small UI change that allows user to quickly alter important manticore options
image

Options presented based on the run-dialog in binja plugin.
Validation is handled by manticore itself, where if an option is invalid the stack trace is printed to the MUI log.

### CODE REVIEW

**Comment by sschriner** [comment]
Closes https://github.com/trailofbits/MUI/issues/29

File: MUI/src/main/java/mui/MUIRunner.java [hunk 1]
```diff
@@ -23,11 +27,33 @@ public void stopProc() {
...
+
+			String optType =
+				(String) MUISettings.SETTINGS.get("NATIVE_RUN_SETTINGS").get(name)[0].get("type");
+
+			if (optType == "string" || optType == "number") {
+
+			}
+			else if (optType == "array") {
+				//TODO: implement
+			}
```
Comment (ekilmer) [inline MUI/src/main/java/mui/MUIRunner.java]: Maybe throw some real warning messages or even exceptions?

File: MUI/src/main/java/mui/MUIRunner.java [hunk 2]
```diff
@@ -23,11 +27,33 @@ public void stopProc() {
 		isTerminated = true;
 	}
 
-	public void callProc(String[] manticoreArgs) {
+	public void constructCommand(String manticoreExePath,
```
Comment (ekilmer) [inline MUI/src/main/java/mui/MUIRunner.java]: This doesn't seem to be used anywhere? Looks like `buildFormPanel` has some similar logic

File: MUI/src/main/java/mui/MUISetupProvider.java [hunk 3]
```diff
@@ -43,61 +53,110 @@ private void setLogProvider(MUILogProvider log) {
...
+				entry.setText(prop.get("default").toString());
+				entry.setToolTipText("You can space-separate multiple arguments");
+				formPanel.add(entry);
+			}
+			else {
+				// TODO: to achieve parity with Binja MUI, type==boolean must be supported, but not needed as part of sensible defaults for running manticore on native binaries
+				throw new UnsupportedOperationException(
+					String.format("[ERROR] Cannot create input row for %s with the type %s", name,
+						prop.get("type")));
+			}
```
Comment (ekilmer) [inline MUI/src/main/java/mui/MUISetupProvider.java]: Would probably be easier to read if you created functions that modified `entry` for each of the respective property types

**Review by ekilmer — CHANGES_REQUESTED** [review]
Looking good! I forget if the Binja GUI plugin has this, but it would be nice to have a text box for "extra Manticore arguments", similar to the original screen but also in addition to some sensible default values and options.

Only some minor nitpicks to address. Once those are done, I think it'll be okay to merge!

File: MUI/src/main/java/mui/MUIRunner.java (hunk 2 above)
Comment (kokrui) [reply MUI/src/main/java/mui/MUIRunner.java]: !! forgot to remove that

**Comment by kokrui** [comment]
resolved the issues and added the "extra mcore args" field!

**Review by ekilmer — APPROVED** [review]
LGTM!

**Pull request merged**